### PR TITLE
AgentCore integration test: prematin startup and agentmain startup test, dynamic configuration zookeeper test, static configuration loading test

### PR DESCRIPTION
**Pull request by TangLeDaily**
【Fix issue】#1298

【Modified content】
1. Added automated integration tests for premain startup and agentmain startup
2. Added automated integration testing for dynamically configuring zookeeper (there is a problem in the Kie test, and only a small amount of code needs to be changed after it is fixed)
3. Added automated integration testing for static configuration loading

[Use case description] Agentcore related test cases

[Self-test status] Local static check passed, local automated test passed

[Scope of influence] None

### CODE REVIEW

File: sermant-integration-tests/agentcore-test/agentcore-integration-test/src/test/java/com/huaweicloud/agentcore/test/request/RequestTest.java [hunk 1]
```diff
@@ -72,4 +74,52 @@ public void testReInstallPlugin() throws IOException {
...
+    /**
+     * 启动后backend的appType和service字段设置测试方法
+     */
+    @Test
+    @EnabledIfSystemProperty(named = "agentcore.test.type", matches = "PREMAIN_STARTUP|AGENTMAIN_STARTUP")
+    public void testBackend() throws IOException {
+        Map<String, Object> resultMap = RequestUtils.analyzingRequestBackend(
+                "http://127.0.0.1:8900/sermant/getPluginsInfo");
+        Assertions.assertNotNull(resultMap, "getPluginsInfo result");
+        Assertions.assertTrue(resultMap.containsKey("appType"), "the result contains appType");
```
Comment (daizhenyu) [inline sermant-integration-tests/agentcore-test/agentcore-integration-test/src/test/java/com/huaweicloud/agentcore/test/request/RequestTest.java]: 抛出的信息写反了吧。下面也排查一下

Comment (TangLeDaily) [reply sermant-integration-tests/agentcore-test/agentcore-integration-test/src/test/java/com/huaweicloud/agentcore/test/request/RequestTest.java]: 已修改

**Comment by luanwenfei-venus** [comment]
去掉描述中的 issue #1236

File: sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/router/ControllerHandler.java [hunk 2]
```diff
@@ -91,6 +91,12 @@ private String setResponse(HttpExchange exchange) {
                 return testController.testUninstallAgent();
             case RouterPath.REQUEST_PATH_REINSTALL_AGENT:
                 return testController.testReInstallAgent();
+            case RouterPath.REQUEST_PATH_PREMAIN:
+                return testController.testPremain();
+            case RouterPath.REQUEST_PATH_AGENTMAIN:
+                return testController.testAgentmain();
```
Comment (luanwenfei-venus) [inline sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/router/ControllerHandler.java]: 建议testPremain和testAgentMain命名更详尽一些，测试用例里的方法可读性很重要

File: sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/router/ControllerHandler.java [hunk 3]
```diff
@@ -91,6 +91,12 @@ private String setResponse(HttpExchange exchange) {
...
+            case RouterPath.REQUEST_PATH_CONFIG_LOAD:
+                return testController.testConfigLoad();
```
Comment (luanwenfei-venus) [inline sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/router/ControllerHandler.java]: ConfigLoad这里详细生命是核心配置和插件配置

File: sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/router/RouterPath.java [hunk 4]
```diff
@@ -53,6 +53,21 @@ public class RouterPath {
...
 
+    /**
+     * 测试premain启动请求地址
+     */
+    public static final String REQUEST_PATH_PREMAIN = "/testPremain";
+
+    /**
+     * 测试agentmain启动请求地址
+     */
+    public static final String REQUEST_PATH_AGENTMAIN = "/testAgentmain";
```
Comment (luanwenfei-venus) [inline sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/router/RouterPath.java]: 同上，建议对应修改

File: sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/tests/dynamic/DynamicTest.java [hunk 5]
```diff
@@ -25,6 +25,12 @@
  * @since 2023-09-08
  */
 public class DynamicTest {
+    /**
+     * 测试的boolean数组下标
+     */
+    private static final int INDEX_0 = 0;
+    private static final int INDEX_1 = 1;
+    private static final int INDEX_2 = 2;
```
Comment (luanwenfei-venus) [inline sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/tests/dynamic/DynamicTest.java]: 这里三个静态常量的命名，我理解应该是指的三种行为的标记吧，这里的命名可以详细写出这三个INDEX所代表的是什么行为标记所在的Index

Comment (TangLeDaily) [reply sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/tests/dynamic/DynamicTest.java]: 已修改

File: sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/router/RouterPath.java (hunk 4 above)
Comment (TangLeDaily) [reply sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/router/RouterPath.java]: 已修改

File: sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/router/ControllerHandler.java (hunk 3 above)
Comment (TangLeDaily) [reply sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/router/ControllerHandler.java]: 已修改

File: sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/router/ControllerHandler.java (hunk 2 above)
Comment (TangLeDaily) [reply sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/router/ControllerHandler.java]: 已修改

**Review by daizhenyu — APPROVED** [review]

**Review by luanwenfei-venus — APPROVED** [review]

**Pull request merged**